AutoNode/validator.py

```diff
@@ -0,0 +1,170 @@
+                                   f'--website "{validator_config["website"]}" '
+                                   f'--security-contact "{validator_config["security-contact"]}" '
+                                   f'--details "{validator_config["details"]}" '
+                                   f'--rate {validator_config["rate"]} '
+                                   f'--max-rate {validator_config["max-rate"]} '
+                                   f'--max-change-rate {validator_config["max-change-rate"]} '
+                                   f'--min-self-delegation {validator_config["min-self-delegation"]} '
+                                   f'--max-total-delegation {validator_config["max-total-delegation"]} '
+                                   f'--amount {validator_config["amount"]} '
+                                   f'--bls-pubkeys {",".join(node_config["public-bls-keys"])} '
+                                   f'--passphrase-file "{saved_wallet_pass_path}" '
+                                   f'--bls-pubkeys-dir "{bls_key_dir}" ')
+        print(f"{Typgpy.OKBLUE}Created Validator!\n{Typgpy.OKGREEN}{response}{Typgpy.ENDC}")
+    except (json.decoder.JSONDecodeError, RuntimeError) as e:
+        print(f"{Typgpy.FAIL}Failed to edit validator!\n\tError: {e}{Typgpy.ENDC}")
+
+
+def _setup(recover_interaction):
+    wait_for_node_response(node_config['endpoint'], verbose=True)
+    all_val_address = get_all_validator_addresses(node_config['endpoint'])  # Check BLS key with validator if it exists
+    if validator_config['validator-addr'] in all_val_address:
+        print(f"{Typgpy.WARNING}{validator_config['validator-addr']} already in list of validators!{Typgpy.ENDC}")
+        if recover_interaction \
+                or input_with_print("Add BLS key(s) to existing validator? [Y]/n \n> ") in {'Y', 'y', 'yes', 'Yes'}:
+            print(f"{Typgpy.HEADER}{Typgpy.BOLD}Editing validator...{Typgpy.ENDC}")
+            _add_bls_key_to_validator()
+    elif validator_config['validator-addr'] not in all_val_address:
+        if recover_interaction \
+                or input_with_print("Create validator? [Y]/n \n> ") in {'Y', 'y', 'yes', 'Yes'}:
+            print(f"{Typgpy.HEADER}{Typgpy.BOLD}Creating new validator...{Typgpy.ENDC}")
+            _create_new_validator()
+    else:
+        node_config['no-validator'] = True
+
+
+def setup(recover_interaction=False, end_sleep=10):
+    if node_config['no-validator']:
+        print(f"{Typgpy.WARNING}Node config specifies not validator automation, exiting...{Typgpy.ENDC}")
+        return
+
+    print(f"{Typgpy.OKBLUE}Create validator config: {Typgpy.OKGREEN}"
+          f"{json.dumps(validator_config, indent=4)}{Typgpy.ENDC}")
+    print(f"{Typgpy.OKBLUE}Using BLS key(s): {Typgpy.OKGREEN}{node_config['public-bls-keys']}{Typgpy.ENDC}")
+    try:
+        _setup(recover_interaction)
+        time.sleep(end_sleep)
+    except Exception as e:
+        time.sleep(end_sleep)
```

Comment:
what is purpoe of sleep in the exception block

Reply:
None, it was some old debugging code that I forgot to remove